### ISSUE TITLE
Emit an event when a peer's subscriptions change

### ISSUE THREAD

**Issue by dirkmc** [libp2p/js-libp2p-floodsub#60]
Some applications that use floodsub as a library need to find out when a peer's subscriptions change (eg see [this peer-star-app issue](https://github.com/ipfs-shipyard/peer-star-app/pull/75#issuecomment-441310497)). It would be helpful to find out immediately, rather than having to poll. I propose emitting an event [when the subscriptions change](https://github.com/libp2p/js-libp2p-floodsub/blob/master/src/index.js#L83).

I'm happy to submit a PR, but I have a question first: I noticed that floodsub [emits events](https://github.com/libp2p/js-libp2p-floodsub/blob/master/src/index.js#L113) with the event name <topic> and the event payload <message>, eg `emit('my topic', <message>)`.

How would you suggest that we name the event so as to avoid clashing with a topic name?

**Comment by vasco-santos** [comment]
I agree that replacing the polling by an event will be more efficient, as well as easy to use for the applications using this implementation. I would really like a PR with this.

Regarding the event name, it has to be good enough to avoid clashing with a topic, as you say. I do not have any strong opinion on a name, except that it should be explicit to its functionality and have a considerable length. Do you have any suggestion? The main thing we have to do is to guarantee that it is really highlighted in the documentation.

**Comment by dirkmc** [comment]
@vasco-santos I made a PR: https://github.com/libp2p/js-libp2p-floodsub/pull/61
